src/ast/schemaPredicates.test.ts

```diff
@@ -29,4 +29,8 @@ describe('SchemaPredicates', () => {
     expect(schemaPredicates.isFieldNullable('Todo', 'complete')).toBeTruthy();
     expect(schemaPredicates.isFieldNullable('Todo', 'author')).toBeTruthy();
   });
+
+  it('should totally do that thing that Steven is struggling with', () => {
+    expect(true).toBeTruthy();
```

Comment:
So we are checking here whether we may have a `UnionType`: https://github.com/FormidableLabs/urql-exchange-graphcache/blob/6e0f016beadf224249cfb5d9362ca979b484fc24/src/ast/schemaPredicates.ts#L98

You can probably add a case here that checks:

```
schemaPredicates.isInterfaceOfType('Todo', 'NoTodosError');
schemaPredicates.isInterfaceOfType('NoTodosError', 'Todo')
```

But I expect our bug to be here: https://github.com/FormidableLabs/urql-exchange-graphcache/blob/6e0f016beadf224249cfb5d9362ca979b484fc24/src/ast/schemaPredicates.ts#L47

We're assuming that `typeCondition` (so `... on NoTodosError`) to be an abstract type when the check fails, but for unions it's actually a regular object type, so we'd have to also check whether we just get the same `ObjectType`, so:

```diff
    const abstractType = this.schema.getType(typeCondition);
+    if (abstractType instanceof GraphQLObjectType) {
+      return abstractType === objectType;
+    }
    expectAbstractType(abstractType, typeCondition);
    const objectType = this.schema.getType(typename);
    expectObjectType(objectType, typename);
```